src/app/frontend/deploy/deployfromsettings.html

```diff
@@ -17,18 +17,21 @@
 <kd-help-section>
   <md-input-container class="md-block" md-is-error="ctrl.isNameError()">
     <label>App name</label>
-    <input ng-model="ctrl.name" name="name" namespace="ctrl.namespace" required kd-unique-name
-        ng-model-options="{ updateOn: 'default blur', debounce: { 'default': 500, 'blur': 0 } }">
-    <md-progress-linear class="kd-deploy-form-progress" md-mode="indeterminate"
-        ng-show="ctrl.form.name.$pending">
-    </md-progress-linear>
-    <ng-messages for="ctrl.form.name.$error" role="alert" multiple>
-      <ng-message when="required">Application name is required.</ng-message>
-      <ng-message when="uniqueName">
-        Application with this name already exists within namespace <i>{{ctrl.namespace}}</i>.
-      </ng-message>
-    </ng-messages>
+    <input ng-model="ctrl.name" name="name" namespace="ctrl.namespace" required ng-pattern="ctrl.namePattern"
+           ng-model-options="{ updateOn: 'default blur', debounce: { 'default': 500, 'blur': 0 } }" kd-unique-name>
+    <div class="progress-or-message-container">
+      <md-progress-linear class="kd-deploy-form-progress" md-mode="indeterminate" ng-show="ctrl.form.name.$pending">
+      </md-progress-linear>
+      <ng-messages for="ctrl.form.name.$error"
+                   role="alert">
+        <ng-message when="required">Application name is required.</ng-message>
+        <ng-message when=="uniqueName">Application with this name
+          already exists within namespace <i>{{ctrl.namespace}}</i>.</ng-message>
+        <ng-message when="pattern">lowercase letters only, '-' between words</ng-message>
```

Comment:
Name validation does not work for me. Something is broken here. Please open with gulp serve:prod and verify.